lib/service/servicecfg/app.go

```diff
@@ -173,6 +190,55 @@ func (a *App) CheckAndSetDefaults() error {
 			}
 		}
 	}
+
+	if len(a.Ports) != 0 {
+		if err := a.checkPorts(); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+func (a *App) checkPorts() error {
```

Comment:
I don't like the fact that this is just a copy of what's already in `api/types/app.go`. But I also noticed that we already duplicate a bunch of stuff between those two locations.

Should I instead create `api/types.ValidateAppPorts` and create an interface with `GetPort` and `GetEndPort` so that I can reuse a single function between `api/types` and `lib/service/servicecfg`?

Reply:
I would move it to somewhere else and re-use it.
Eg `api/utils/net.ValidatePortRange(port, portEnd int) error`